File: Tareas/tarea16.rb
puts "Azul Carolina Martínez Cabañas"
puts "1868435					 ITS"
puts "Periodo Enero-Junio 2020"
puts "Ing. Ismael Gutiérrez Pimentel"
puts "Multiplicación de Matrices"
print "Ingresar el número de filas de la matriz A: "
mA = gets().to_i
print "Ingresar el número de columnas de la matriz A: "
nA = gets().to_i
print "Ingresar el número de filas de la matriz B: "
mB = gets().to_i
print "Ingresar el número de columnas de la matriz B: "
nB = gets().to_i
if nA == mB
    matA = Array.new(mA){Array.new(nA)}
    matB = Array.new(mB){Array.new(nB)}
    multi = Array.new(mA){Array.new(nB)}
    for i in(0..(mA - 1))
        for j in(0..(nA - 1))
            print "Ingresar los números que pertenecen a la matriz A[#{i+1}][#{j+1}] :"
            matA[i][j]=gets().to_i
        end
    end
    for i in(0..(mB - 1))
        for j in(0..(nB - 1))
            print "Ingresar los números que pertenecen a la matriz B[#{i+1}][#{j+1}] :"
            matB[i][j]=gets().to_i
        end
    end
	display=""
	for i in(0...mA)
    	for j in(0...nB)
			multi[i][j] = 0
			for k in(0...nA)
    			multi[i][j] = (matA[i][k] * matB[k][j]) + multi[i][j]
			end
			display = display + multi[i][j].to_s + " "
		end
		display = display + "\n"
	end
	puts "El resultado de la multiplicación de las matrices es: "
	puts(display)
else
    abort "Las dimensiones de las matrices no son válidas"
end

File: Tareas/tarea16.java
import javax.swing.*;


public class Tarea16 {
    public static void main(String[] args) {
        JOptionPane.showMessageDialog(null, "Azul Carolina Martínez Cabañas\n1868435 ITS\nPeriodo Enero-Junio 2020\nIng. Ismael Gutiérrez Pimentel");
        int mA = Integer.parseInt(JOptionPane.showInputDialog("Multiplicación de Matrices\nIngresar el número de filas de la matriz A:"));
        int nA = Integer.parseInt(JOptionPane.showInputDialog("Ingresar el número de columnas de la matriz A:"));
        int mB = Integer.parseInt(JOptionPane.showInputDialog("Ingresar el número de filas de la matriz B:"));
        int nB = Integer.parseInt(JOptionPane.showInputDialog("Ingresar el número de columnas de la matriz B:"));
        if(nA == mB){
            int [][] matA = new int[mA][nA];
            int [][] matB = new int[mB][nB];
            int [][] multi = new int[mA][nB];
            for (int i = 0; i <=(mA-1) ; i++) {
                for (int j = 0; j <=(nA -1); j++) {
                    matA[i][j] = Integer.parseInt(JOptionPane.showInputDialog("Ingresar los números que pertenecen a la matriz A "));
                }
            }
            for (int i = 0; i <=(mB -1) ; i++) {
                for (int j = 0; j <= (nB -1); j++) {
                    matB[i][j] = Integer.parseInt(JOptionPane.showInputDialog("Ingresar los números que pertenecen a la matriz B"));
                }
            }
            String display ="";
            for (int i = 0; i <mA ; i++) {
                for (int j = 0; j <nB ; j++) {
                    multi[i][j] = 0;
                    for (int k = 0; k < nA ; k++) {
                        multi[i][j] = (matA[i][k] * matB[k][j]) + multi[i][j];
                    }
                    display += multi[i][j] + " ";
                }
                display += "\n";
            }
            JOptionPane.showMessageDialog(null, "El resultado de la multiplicación de matrices es: \n"+display);
        }else{
            JOptionPane.showMessageDialog(null, "Las dimensiones no son válidas");
            System.exit(0);
        }
    }
}